libbeat/kibana/client.go

```diff
@@ -269,33 +281,48 @@ func (client *Client) readVersion() error {
 // IgnoreVersion was set when creating the client.
 func (client *Client) GetVersion() common.Version { return client.Version }
 
-func (client *Client) ImportJSON(url string, params url.Values, jsonBody map[string]interface{}) error {
+func (client *Client) ImportMultiPartFormFile(url string, params url.Values, filename string, contents string) error {
+	buf := &bytes.Buffer{}
+	w := multipart.NewWriter(buf)
+
+	pHeaders := textproto.MIMEHeader{}
+	pHeaders.Add("Content-Disposition", fmt.Sprintf(`form-data; name="file"; filename="%s"`, filename))
+	pHeaders.Add("Content-Type", "application/ndjson")
 
-	body, err := json.Marshal(jsonBody)
+	p, err := w.CreatePart(pHeaders)
 	if err != nil {
-		client.log.Debugf("Failed to json encode body (%v): %#v", err, jsonBody)
-		return fmt.Errorf("fail to marshal the json content: %v", err)
+		return fmt.Errorf("failed to create multipart writer for payload: %+v", err)
 	}
-
-	statusCode, response, err := client.Connection.Request("POST", url, params, nil, bytes.NewBuffer(body))
+	_, err = io.Copy(p, strings.NewReader(contents))
 	if err != nil {
-		return fmt.Errorf("%v. Response: %s", err, truncateString(response))
+		return fmt.Errorf("failed to copy contents of the object: %+v", err)
 	}
-	if statusCode >= 300 {
+	w.Close()
+
+	headers := http.Header{}
+	headers.Add("Content-Type", w.FormDataContentType())
+
+	statusCode, response, err := client.Connection.Request("POST", url, params, headers, buf)
+	if err != nil || statusCode >= 300 {
 		return fmt.Errorf("returned %d to import file: %v. Response: %s", statusCode, err, response)
 	}
+
+	client.log.Debugf("Imported multipart file to %s with params %v", url, params)
 	return nil
 }
 
 func (client *Client) Close() error { return nil }
 
 // GetDashboard returns the dashboard with the given id with the index pattern removed
-func (client *Client) GetDashboard(id string) (common.MapStr, error) {
-	params := url.Values{}
-	params.Add("dashboard", id)
-	_, response, err := client.Request("GET", "/api/kibana/dashboards/export", params, nil, nil)
-	if err != nil {
-		return nil, fmt.Errorf("error exporting dashboard: %+v", err)
+func (client *Client) GetDashboard(id string) ([]byte, error) {
+	if client.Version.LessThanOrEqual(true, common.MustNewVersion("7.14.0")) {
```

Comment:
There appear to be several references to this version. Consider making this version requirement a constant somewhere.